rest-api/src/main/java/com/cloudbees/workflow/rest/external/RunExt.java

```diff
@@ -332,8 +332,8 @@ public static RunExt createOld(WorkflowRun run) {
             if (artifacts != null && !artifacts.isEmpty()) {
                 runExt.get_links().setArtifacts(Link.newLink(RunAPI.getArtifactsUrl(run)));
             }
-
             FlowGraphWalker walker = new FlowGraphWalker(execution);
+            List<FlowNode> sortedNodes = FlowNodeUtil.getIdSortedExecutionNodeList(execution); // Hold a ref to prevent GC until we're done analyzing
```

Comment:
Well as of JDK 8u74 if you comment out [this line](https://github.com/jenkinsci/jenkins/blob/511ccf0c836651d1d825c062d7d42338438c735c/test/src/test/java/hudson/diagnosis/OldDataMonitorTest.java#L101) the test fails, which I take as evidence that at least the first-pass compiler is _not_ that aggressive. Whether the optimized version produced after you run the method a few thousand times tries to discard the reference, I do not know.